File: backend/internal/delivery/http/dto/ticket.go
package dto

import (
	"time"

	"github.com/cassianobraz/SistemaControleChamados/backend/internal/domain"
)

type CreateTicketRequest struct {
	Title         string `json:"title" example:"Impressora do 2º andar não funciona"`
	Description   string `json:"description" example:"A impressora está travando toda vez que alguém tenta imprimir."`
	Priority      string `json:"priority" example:"media"`
	ResponsibleID string `json:"responsible_id,omitempty" example:""`
}

type UpdateTicketRequest struct {
	Title       string `json:"title" example:"Impressora do 2º andar não funciona"`
	Description string `json:"description" example:"A impressora está travando toda vez que alguém tenta imprimir."`
	Priority    string `json:"priority" example:"alta"`
	Status      string `json:"status" example:"em_andamento"`
}

type AssignTicketRequest struct {
	ResponsibleID string `json:"responsible_id,omitempty" example:""`
}

type TicketResponse struct {
	ID              string    `json:"id"`
	Title           string    `json:"title"`
	Description     string    `json:"description"`
	Priority        string    `json:"priority"`
	Status          string    `json:"status"`
	ResponsibleID   string    `json:"responsible_id"`
	ResponsibleName string    `json:"responsible_name"`
	CreatedAt       time.Time `json:"created_at"`
	UpdatedAt       time.Time `json:"updated_at"`
}

func NewTicketResponse(ticket *domain.Ticket) TicketResponse {
	return TicketResponse{
		ID:              ticket.ID,
		Title:           ticket.Title,
		Description:     ticket.Description,
		Priority:        ticket.Priority.String(),
		Status:          ticket.Status.String(),
		ResponsibleID:   ticket.ResponsibleID,
		ResponsibleName: ticket.ResponsibleName,
		CreatedAt:       ticket.CreatedAt,
		UpdatedAt:       ticket.UpdatedAt,
	}
}

func NewTicketResponseList(tickets []*domain.Ticket) []TicketResponse {
	responses := make([]TicketResponse, 0, len(tickets))
	for _, ticket := range tickets {
		responses = append(responses, NewTicketResponse(ticket))
	}

	return responses
}

type TicketListResponse struct {
	Data       []TicketResponse `json:"data"`
	Total      int64            `json:"total"`
	Page       int              `json:"page"`
	PageSize   int              `json:"page_size"`
	TotalPages int64            `json:"total_pages"`
}

File: frontend/src/types/ticket.ts
export type Priority = 'baixa' | 'media' | 'alta'

export type Status = 'aberto' | 'em_andamento' | 'resolvido' | 'fechado'

export type SortOrder = 'created_at_desc' | 'created_at_asc' | 'priority_desc'

export interface Ticket {
  id: string
  title: string
  description: string
  priority: Priority
  status: Status
  responsible_id: string
  responsible_name: string
  created_at: string
  updated_at: string
}

export interface TicketListResponse {
  data: Ticket[]
  total: number
  page: number
  page_size: number
  total_pages: number
}

export interface CreateTicketPayload {
  title: string
  description: string
  priority: Priority
  responsible_id?: string
}

export interface UpdateTicketPayload {
  title: string
  description: string
  priority: Priority
  status: Status
}

export interface AssignTicketPayload {
  responsible_id?: string
}

export interface ListTicketsParams {
  status?: Status
  priority?: Priority
  responsible_id?: string
  search?: string
  sort?: SortOrder
  page?: number
  page_size?: number
}
